src/components/Checkbox/Checkbox.tsx

```diff
@@ -21,64 +18,59 @@ export class Checkbox extends React.Component<ICheckboxProps, ICheckboxState> {
   constructor(props: ICheckboxProps) {
     super(props);
 
-    this.state = {
-      isChecked: props.isChecked
-    };
-
-    this._id = `checkbox-${ _instance++ }`;
-
-    this._handleInputChange = this._handleInputChange.bind(this);
-  }
-
-  public componentWillReceiveProps(newProps: ICheckboxProps) {
-    if (newProps.isChecked !== this.props.isChecked) {
-      this.setState({
-        isChecked: newProps.isChecked
-      });
-    }
+    this._id = getId('checkbox-');
+    this._onChange = this._onChange.bind(this);
   }
 
   public render() {
-    let { text, isEnabled, className } = this.props;
-    let { isChecked } = this.state;
+    const {
+      checked,
+      className,
+      defaultChecked,
+      disabled,
+      inputProps,
+      label
+    } = this.props;
 
     return (
       <div className={ css('ms-ChoiceField', className) }>
         <input
-          ref={ (c): HTMLInputElement => this._checkBox = c }
+          { ...inputProps }
+          { ...(checked !== undefined && { checked }) }
+          { ...(defaultChecked !== undefined && { defaultChecked }) }
+          disabled={ disabled }
+          ref={ (el): HTMLInputElement => this._checkBox = el }
           id={ this._id }
           name={ this._id }
           className='ms-ChoiceField-input'
           type='checkbox'
           role='checkbox'
-          checked={ isChecked }
-          disabled={ !isEnabled }
-          onChange={ this._handleInputChange }
-          aria-checked={ isChecked }
+          onChange={ this._onChange }
+          aria-checked={ checked }
         />
         <label htmlFor={ this._id } className='ms-ChoiceField-field'>
-          { text && <span className='ms-Label'>{ text }</span> }
+          { label && <span className='ms-Label'>{ label }</span> }
         </label>
       </div>
     );
   }
 
+  public get checked(): boolean {
+    return this._checkBox ? this._checkBox.checked : false;
+  }
+
   public focus() {
       if (this._checkBox) {
           this._checkBox.focus();
       }
   }
 
-  private _handleInputChange(evt: React.FormEvent) {
-    let { onChanged } = this.props;
-    const isChecked = (evt.target as HTMLInputElement).checked;
-
-    this.setState({
-      isChecked: isChecked
-    });
+  private _onChange(ev: React.FormEvent) {
+    const { onChange } = this.props;
+    const isChecked = (ev.target as HTMLInputElement).checked;
```

Comment:
if you pass in a value, you are responsible for handling updates to that value. `isChecked` will actually be false in the `_onChange` handler and if you decide not to update the state with the new value, it will render as `true` again. So it's actually working as intended